config.example.yaml

```diff
@@ -76,6 +76,8 @@ logging:
   instance_name: block-header-service
   # Flag for enabling origin parameter in logs
   origin: false
+  # Release mode for Gin web framework should be enabled in "production"
+  gin_release_mode: false
```

Comment:
I wonder if we really need the ability to set `gin_mode` separately from the login level. Maybe it's enough to assume that `log.level == debug => gin_mode == debug`?

Reply:
If I understand correctly, it's not about the log level. It's about production/development. If someone wants to have `log.level: debug` in production, he shouldn't automatically have `gin_mode` in `debug`

Reply:
but if you set gin_mode to Release it will not generate debug informations, on the other hand if you set gin to debug mode but your logger will be set to eg. Info level you won't see them either

Reply:
As I looking into the `gin` repo. They use it only `to print/or not to print` internal debug messages:
```golang
func IsDebugging() bool {
	return ginMode == debugCode
}
```
https://github.com/gin-gonic/gin/blob/ecdbbbe9483dd12222f2085f717a2c7cb5ac55fe/debug.go#L19C6-L19C17

So maybe @arkadiuszos4chain has a good point to set gin_release_mode when `logLevel >= INFO` 🤔

Reply:
Let me ask this question differently: do we need the ability to set a dependent module (such as gin) in a different mode than the main application?

Reply:
Maybe, when we do not wan't to collect huge amount of logs from entire app but we'd like to "debug" a problem with gin which occurs only at production (with avarage/high load).

But as I'm rethinking this issue, I would agree with @arkadiuszos4chain suggestion

Reply:
After a conversation with Arek, I've removed the new config parameter and made the Gin release mode dependent on the log level. Please re-review.